src/controllers/userController.js

```diff
@@ -123,8 +125,87 @@ class UserController {
       });
     }
   }
+
+  /**
+   * @description - This method is responsible for assigning roles to users
+   *
+   * @static
+   * @param {object} request - Request sent to the router
+   * @param {object} response - Response sent from the controller
+   *
+   * @returns {object} - object representing response message
+   *
+   * @memberof UserController
+   */
+  static async assignRole(request, response) {
+    try {
+      const { email, role } = request.body;
+      if (role !== 'admin' && role !== 'user') {
+        return response.status(400).json({
+          status: 'Fail',
+          message: 'Status can only be "admin" or "user"'
+        });
+      }
+      const fieldToUpdate = {
+        role
+      };
+      const updatedUser = await User.update(fieldToUpdate, {
+        where: {
+          email
+        },
+        returning: true
+      });
+      return response.status(200).json({
+        status: 'Success',
+        message: 'User role updated successfully',
+        updatedUser
+      });
+    } catch (error) {
+      response.status(500).json({
+        status: 'Fail',
+        message: errorResponse(['internal server error, please try again later']),
+      });
+    }
+  }
+
+  /**
+   * @description - This method is responsible for deleting users
+   *
+   * @static
+   * @param {object} request - Request sent to the router
+   * @param {object} response - Response sent from the controller
+   *
+   * @returns {object} - object representing response message
+   *
+   * @memberof UserController
+   */
+  static async deleteUser(request, response) {
+    const { email } = request.body;
+    try {
+      await User.destroy({ where: { email } });
```

Comment:
Do you think we should outright delete the user from the database? Why the `destroy`?

Reply:
That right is as you can see is reserved for the admin and super admin. e.g When a user abuses the platform several times the admin or super admin can delete the user.